device-to-group-command.py

```diff
@@ -0,0 +1,404 @@
+    #print(command)
+    try:
+        api_response = api_instance.run_command(enterprise_id, device.id, command)
+        if api_response.id is not None:
+            print('Uninstall command fired successfully on ' +  device.device_name + ' for ' + package_name)
+    except ApiException as e:
+        print("Exception when calling CommandsApi->run_command: %s\n" % e)
+
+
+def uninstall_package_in_group_devices(device_list, package_name):
+    # Fetch all devices in group
+
+    print("Currently active devices in group: ")
+    print([device.device_name for device in device_list])
+
+    for device in device_list:
+        app_id = is_package_installed(device.id, package_name)
+        if app_id:
+            try:
+                run_uninstall_command(device, package_name, app_id)
+            except Exception as e:
+                print('Could not fire command on ' + device.device_name + ": " + str(e))
+            time.sleep(1)
+        else:
+            print('Warning: Package already uninstalled or not present on device: ' + device.device_name)
+
+def is_package_installed(device_id, package_name):
+    querystring = {"package_name": package_name, "whitelisted":"false"}
+    payload = ""
+    headers = {
+        'Authorization': "Bearer " + configuration.api_key['Authorization'],
+        'cache-control': "no-cache",
+        }
+
+    api_instance = esperclient.DeviceApi(esperclient.ApiClient(configuration))
+    response = api_instance.get_app_installs(enterprise_id, device_id, package_name=package_name)
+    #print(response)
+    if response and response.count > 0:
+        app_id = response.results[0].application.version.app_version_id
+        if app_id:
+          return response.results[0].application.version.app_version_id
+        else:
+            return 0
+    else:
+        #print("App doesn't exist")
+        return 0
+
+
+# ====== Get All the devices in a group ========
+
+def get_devices_in_group(group_id):
+    # create an instance of the API class
+
+    api_instance = esperclient.DeviceApi(esperclient.ApiClient(configuration))
+    limit = 5000 # int | Number of results to return per page. (optional) (default to 20)
+    offset = 0 # int | The initial index from which to return the results. (optional) (default to 0)
+
+    try:
+        api_response = api_instance.get_all_devices(enterprise_id, group=group_id, limit=limit, offset=offset)
+        #print(api_response)
+        active_device_list = []
+        # None check
+        if len(api_response.results) > 0:
+            for device in api_response.results:
+                if device.status == 1:  # Check for active devices only
+                    active_device_list.append(device)
+                else:
+                    inactive_device_list.append(device)
+    except ApiException as e:
+        print("Exception when calling DeviceApi->get_all_devices: %s\n" % e)
+
+    return active_device_list
```

Comment:
this will return None if exception is thrown.

Reply:
done